cmd/kubeadm/app/cmd/phases/addons.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"github.com/spf13/cobra"
+
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation"
+	dnsaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns"
+	proxyaddon "k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/proxy"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	kubeconfigutil "k8s.io/kubernetes/cmd/kubeadm/app/util/kubeconfig"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// NewCmdAddon returns the addon Cobra command
+func NewCmdAddon() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "addon <addon-name>",
+		Aliases: []string{"addons"},
+		Short:   "Install an addon to a Kubernetes cluster.",
+		RunE:    subCmdRunE("addon"),
+	}
+
+	cmd.AddCommand(getAddonsSubCommands()...)
+	return cmd
+}
+
+func EnsureAllAddons(cfg *kubeadmapi.MasterConfiguration, client clientset.Interface) error {
```

Comment:
Is this function going to be replaced with https://github.com/kubernetes/kubernetes/pull/51171/files#diff-eadad99fe8edbb21ce1c0f32a362240a?

Reply:
Yes, I will make that change and have it up in a bit.